lib/simplifyExpression/basicsSearch/removeMultiplicationByNegativeOne.js

```diff
@@ -1,56 +1,30 @@
-const clone = require('../../util/clone');
+const {rewriteNode} = require('math-rules');
 
 const ChangeTypes = require('../../ChangeTypes');
-const Negative = require('../../Negative');
+const defineRuleString = require('../../util/defineRuleString');
 const Node = require('../../node');
 
-// If `node` is a multiplication node with -1 as one of its operands,
-// and a non constant as the next operand, remove -1 from the operands
-// list and make the next term have a unary minus.
+// e.g. x * -1 -> -x
+const REMOVE_MULTIPLYING_BY_NEGATIVE_ONE = defineRuleString('#a * -1', '-#a');
+
+// e.g. -1 * x -> -x
+const REMOVE_MULTIPLYING_BY_NEGATIVE_ONE_REVERSE = defineRuleString('-1 * #a', '-#a');
+
+// If `node` is of the form 1^x, reduces it to a node of the form 1.
 // Returns a Node.Status object.
 function removeMultiplicationByNegativeOne(node) {
-  if (node.op !== '*') {
-    return Node.Status.noChange(node);
+  let newNode = null;
+  newNode = rewriteNode(REMOVE_MULTIPLYING_BY_NEGATIVE_ONE, node);
+  if (!newNode) {
+    newNode = rewriteNode(REMOVE_MULTIPLYING_BY_NEGATIVE_ONE_REVERSE, node);
   }
-  const minusOneIndex = node.args.findIndex(arg => {
-    return Node.Type.isConstant(arg) && arg.value === '-1';
-  });
-  if (minusOneIndex < 0) {
-    return Node.Status.noChange(node);
-  }
-
-  // We might merge/combine the negative one into another node. This stores
-  // the index of that other node in the arg list.
-  let nodeToCombineIndex;
-  // If minus one is the last term, maybe combine with the term before
-  if (minusOneIndex + 1 === node.args.length) {
-    nodeToCombineIndex = minusOneIndex - 1;
+  if (newNode) {
+    return Node.Status.nodeChanged(
+      ChangeTypes.REMOVE_MULTIPLYING_BY_NEGATIVE_ONE, node, newNode);
   }
   else {
-    nodeToCombineIndex = minusOneIndex + 1;
-  }
-
-  let nodeToCombine = node.args[nodeToCombineIndex];
-  // If it's a constant, the combining of those terms is handled elsewhere.
-  if (Node.Type.isConstant(nodeToCombine)) {
     return Node.Status.noChange(node);
   }
-
-  let newNode = clone(node);
-
-  // Get rid of the -1
-  nodeToCombine = Negative.negate(clone(nodeToCombine));
-
-  // replace the node next to -1 and remove -1
-  newNode.args[nodeToCombineIndex] = nodeToCombine;
-  newNode.args.splice(minusOneIndex, 1);
-
-  // if there's only one operand left, move it up the tree
```

Comment:
This is something we should test for explicitly because we want to avoid having a tree where there's a `mul` node with a single argument, e.g.:
```
( + x ( * -1 ) y )
```